.github/actions/is-fork/action.yml

```diff
@@ -0,0 +1,22 @@
+---
+name: Is Fork
+
+description: |
+  Checks if current job runs on code from a forked repository (via PR).
+  This information can be used to disable certain CI features that are not
+  available on forks (e.g. access to secrets).
+
+inputs: {}
+
+outputs:
+  is-fork:
+    description: Output whether the current job runs on code from a forked repository
+    value: ${{ steps.detect-fork.outputs.is-fork == 'true' }}
+
+runs:
+  using: composite
+  steps:
+  - id: detect-fork
+    shell: bash
+    run: |
+      echo is-fork="${{ github.event.pull_request.head.repo.fork }}" | tee -a $GITHUB_OUTPUT
```

Comment:
This value can be empty in some contexts (on push, in merge queue, on schedule) but is caught by the `== 'true'` comparison in line 14.

Reply:
Not blocking and even not problematic here, but I always use intermediate environment variable as a good practice for mitigating script injection attacks, as advised here https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions

Reply:
Personal opinion: it also improves readability :)

Reply:
Yea I should do it as well.